foursight_core/app_utils.py

```diff
@@ -334,6 +547,175 @@ def trim_output(cls, output, max_size=100000):
             return cls.TRIM_ERR_OUTPUT
         return output
 
+    def sorted_dict(self, dictionary: dict) -> dict:
+        result = {}
+        for key in sorted(dict(dictionary).keys(), key=lambda key: key.lower()):
+            result[key] = dictionary[key]
+        return result
+
+    def get_aws_account_number(self) -> dict:
+        try:
+            caller_identity = boto3.client("sts").get_caller_identity()
+            return caller_identity["Account"]
+        except Exception:
+            logger.warn("Cannot determine AWS credentials token (not fatal - just for Foursight UI display).")
+            return None
+
+    def get_obfuscated_credentials_info(self, env_name: str, stage_name: str) -> dict:
+        try:
+            session = boto3.session.Session()
+            credentials = session.get_credentials()
+            access_key_id = credentials.access_key
+            region_name = session.region_name
+            caller_identity = boto3.client("sts").get_caller_identity()
+            user_arn = caller_identity["Arn"]
+            account_number = caller_identity["Account"]
+            auth0_client_id = self.get_auth0_client_id(env_name, stage_name)
+            credentials_info = {
+                "AWS Account Number:": account_number,
+                "AWS User ARN:": user_arn,
+                "AWS Access Key ID:": access_key_id,
+                "AWS Region Name:": region_name,
+                "Auth0 Client ID:": auth0_client_id
+            }
+            return credentials_info
+        except Exception as e:
+            logger.warn("Cannot determine AWS credentials token (not fatal - just for Foursight UI display).")
```

Comment:
This warning is repeated and should be centralized, either by a variable containing the text or by function.

Also, there are potentially many errors that could happen here and not all are AWS credentials problems. The error message seems a lie. I think the warning should include the actual error string. But also, the message itself suggested to me that it was summarizing an error that it couldn't access credentials rather than it couldn't compose credentials info for display. I think I'd like "Failure in get_obfuscated_credentials_info." or (as above) "Failure in get_aws_acount_number." Whatever this is about an account token seems a red herring even if it is the most likely cause. Let the underlying error say that's the problem if indeed it is. If the goal is to keep creds info out of the logs, then again make sure the error summary is precise by saying the function that is failing and what its arguments are, not by trying to guess why.

I think you want something vaguely like:
```
from dcicutils.misc_utils import get_error_message  # put import at top of file
def notice_failed_ui_info_acquisition(error_object, calling_function):
    logger.warn(f"{calling_function} got an error."
                f" This error is not fatal because it only supplies information for the user"
                f" to see in the Foursight UI."
                f" The underlying error was: {get_error_message(error_object)}")
```